core/domain/user_id_migration.py

```diff
@@ -899,9 +1073,106 @@ def reduce(key, ids):
         yield (key, len(ids))
 
 
+class AddAllUserIdsSnapshotMetadataVerificationJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """For every snapshot metadata of a rights model, merge the data from all
+    the user id fields in commit_cmds together and put them in the all_user_ids
+    field of an appropriate RightsAllUsersModel.
+    """
+
+    @staticmethod
+    def _add_col_and_exp_user_ids(snapshot_model, all_users_model):
+        """Merge the user ids from the snapshot and put them in the parent
+        collection rights model.
+        """
+        if are_commit_cmds_role_change(snapshot_model.commit_cmds):
+            all_user_ids = set(all_users_model.all_user_ids)
+            all_user_ids.add(snapshot_model.commit_cmds[0]['assignee_id'])
+            all_users_model.all_user_ids = list(all_user_ids)
```

Comment:
I would argue that we shouldn't impose any order on the fields that contain multiple user IDs. Even tho the fields are list I think we should think about them as sets and these are usually not sorted. The ordering is not used anywhere else and I don't see any benefit in adding it here.

We can maybe discuss this in some call but I think if we want to do this anyway it would make sense to do it in a different PR.

Reply:
Also, ordering of the IDs can prolong the jobs times by a few seconds without any clear benefit.

Reply:
See https://github.com/oppia/oppia/pull/8505#discussion_r426188943

Reply:
I'm not too bothered about the time complexity because the lists are very small and a few seconds is not a big deal.

But it looks like these are long-lived fields, right? If so, then let's please keep them sorted. It is better to have more constraints on the models so that, when reading from them, you can assume more things. (If the models in question were temporary I wouldn't care as much.)

Reply:
Nope, this model will only be used for the migration and remove afterward.